Push notification crash when app is closed

I got a problem with Unity 5.5.5p1 & Android oreo (don't know if it impact more Unity versions or Android versions)

Log :

> 02-10 10:29:14.805 16199 16199 I FirebaseInitProvider: FirebaseApp initialization unsuccessful
> 02-10 10:29:14.839 16199 16199 D AndroidRuntime: Shutting down VM
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime: FATAL EXCEPTION: main
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime: Process: com.company.game, PID: 16199
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime: java.lang.RuntimeException: Unable to start receiver net.agasper.unitynotification.UnityNotificationManager: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.app.Activity.getSystemService(java.lang.String)' on a null object reference
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3259)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at android.app.ActivityThread.-wrap17(Unknown Source:0)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1677)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:105)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6541)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.app.Activity.getSystemService(java.lang.String)' on a null object reference
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at net.agasper.unitynotification.UnityNotificationManager.CreateChannel(UnityNotificationManager.java:41)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at net.agasper.unitynotification.UnityNotificationManager.createChannelIfNeeded(UnityNotificationManager.java:64)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at net.agasper.unitynotification.UnityNotificationManager.onReceive(UnityNotificationManager.java:154)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3252)
> 02-10 10:29:14.850 16199 16199 E AndroidRuntime:        ... 8 more

Even with the test scene.
I solved it (apparently) by trasnfering the argument  Context context from onReceive to CreateChannel : 


        public static void CreateChannel(Context context, String identifier, String name, String description, int importance, String soundName, int enableLights, int lightColor, int enableVibration, long[] vibrationPattern, String bundle) {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O)
            return;

        channels.add(identifier);
        NotificationManager nm = null;
        if(UnityPlayer.currentActivity != null)
        {
                    nm = (NotificationManager) UnityPlayer.currentActivity.getSystemService(Context.NOTIFICATION_SERVICE);
        }
        else if (context != null)
        {
            nm = (NotificationManager)context.getSystemService(Context.NOTIFICATION_SERVICE);
        }


1/ I don't know if it's the way to go
2/ I don't know if it will be efficient in all the cases

But maybe it will help somebody / help me when you'll say that's was mistake from me !

--- a/PluginSrc/app/src/main/java/net/agasper/unitynotification/UnityNotificationManager.java
+++ b/PluginSrc/app/src/main/java/net/agasper/unitynotification/UnityNotificationManager.java
@@ -1,5 +1,6 @@
 package net.agasper.unitynotification;
 
+import android.annotation.TargetApi;
 import android.app.Activity;
 import android.app.AlarmManager;
 import android.app.Notification;
@@ -69,6 +70,12 @@ public class UnityNotificationManager extends BroadcastReceiver
                                        int lights, String largeIconResource, String smallIconResource, int bgColor, String bundle, String channel,
                                        ArrayList<NotificationAction> actions)
     {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            if (channel == null)
+                channel = "default";
+            createChannelIfNeeded(channel, title, soundName, lights == 1, vibrate == 1, bundle);
+        }
+
         Activity currentActivity = UnityPlayer.currentActivity;
         AlarmManager am = (AlarmManager)currentActivity.getSystemService(Context.ALARM_SERVICE);
         Intent intent = new Intent(currentActivity, UnityNotificationManager.class);
@@ -97,6 +104,12 @@ public class UnityNotificationManager extends BroadcastReceiver
     public static void SetRepeatingNotification(int id, long delayMs, String title, String message, String ticker, long rep, int sound, String soundName, int vibrate, int lights,
                                                 String largeIconResource, String smallIconResource, int bgColor, String bundle, String channel, ArrayList<NotificationAction> actions)
     {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            if (channel == null)
+                channel = "default";
+            createChannelIfNeeded(channel, title, soundName, lights == 1, vibrate == 1, bundle);
+        }
+
         Activity currentActivity = UnityPlayer.currentActivity;
         AlarmManager am = (AlarmManager)currentActivity.getSystemService(Context.ALARM_SERVICE);
         Intent intent = new Intent(currentActivity, UnityNotificationManager.class);
@@ -151,9 +164,6 @@ public class UnityNotificationManager extends BroadcastReceiver
 
         if (channel == null)
             channel = "default";
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            createChannelIfNeeded(channel, title, soundName, lights, vibrate, bundle);
-        }
 
         NotificationCompat.Builder builder = new NotificationCompat.Builder(context, channel);
 

